nltk/text.py

```diff
@@ -173,36 +171,57 @@ def __repr__(self):
         return '<ConcordanceIndex for %d tokens (%d types)>' % (
             len(self._tokens), len(self._offsets))
 
-    def print_concordance(self, word, width=75, lines=25):
-        """
-        Print a concordance for ``word`` with the specified context window.
+    def generate_concordance(self, word, width=80):
+        """Generate a concordance list of tuples(left, token, right)
+        for ``word`` with the specified context window.
 
         :param word: The target word
         :type word: str
         :param width: The width of each line, in characters (default=80)
         :type width: int
-        :param lines: The number of lines to display (default=25)
-        :type lines: int
         """
+
+        self._concordance = []
+
         half_width = (width - len(word) - 2) // 2
         context = width // 4 # approx number of words of context
-
         offsets = self.offsets(word)
         if offsets:
-            lines = min(lines, len(offsets))
-            print("Displaying %s of %s matches:" % (lines, len(offsets)))
             for i in offsets:
-                if lines <= 0:
-                    break
                 left = (' ' * half_width +
-                        ' '.join(self._tokens[i-context:i]))
-                right = ' '.join(self._tokens[i+1:i+context])
+                        ' '.join(self._tokens[i - context:i]))
+                right = ' '.join(self._tokens[i + 1:i + context])
                 left = left[-half_width:]
                 right = right[:half_width]
-                print(left, self._tokens[i], right)
-                lines -= 1
+                self._concordance.append((left, self._tokens[i], right))
+
+    def list_concordance(self, lines=None):
+        """
+        :rtype: list(str)
+        :return: list concordances as strings
+        :param length:  The number of concordances to return (default=None)
+        :type length: int
+        """
+        return [" ".join([left, token, right]) for
+                (left, token, right) in self._concordance[:lines]]
+
+    def print_concordance(self, lines=25):
+        """
```

Comment:
I'm kinda confused. generate_concordance is just the name I gave the functionality that was inside [print_concordance](https://github.com/nltk/nltk/blob/develop/nltk/text.py#L176) because it needs to get pulled out to be used for both printing and making the list.